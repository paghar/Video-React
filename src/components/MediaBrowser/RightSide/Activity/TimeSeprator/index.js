//@flow
import React from 'react';
import { getShortDate } from 'helpers/time';
import { Time } from './index.style';
import { Display4 } from 'theme/fonts';
import { Generals } from 'theme/colors';
type Props = {
  time: number,
  className?: string,
};
const oneDay = 1000 * 60 * 60 * 24;
const TimeSeprator = ({ time, className }: Props) => {
  let today = new Date().setHours(0);
  today = new Date(today).setMinutes(0);
  today = new Date(today).setSeconds(0);
  const diff = parseInt(time - today);
  return (
    <Time className={className} font={Display4} hoverable color={Generals.vii}>
      {diff < oneDay && diff > 0
        ? 'Today'
        : diff < 0 && diff > -oneDay
        ? 'Yesterday'
        : getShortDate(time)}
    </Time>
  );
};

export default TimeSeprator;
