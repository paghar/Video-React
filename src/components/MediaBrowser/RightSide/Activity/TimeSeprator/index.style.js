//@flow
import styled from 'styled-components';
import Typography from 'components/shared/Typography';

export const Time = styled(Typography)`
  background: none;
  outline: none;
  border: none;
  display: flex;
  text-align: center;
  align-items: center;
  justify-content: center;
  width: 100%;
`;
