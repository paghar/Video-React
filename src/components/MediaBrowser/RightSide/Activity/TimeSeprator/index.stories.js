//@flow
import React from 'react';
import TimeSeprator from '.';

export default { title: 'TimeSeprator' };

export const time = () => <TimeSeprator time={1570963070000} />;
export const PastFriday = () => <TimeSeprator time={1570771290000} />;
export const LastYear = () => <TimeSeprator time={1543987290000} />;
