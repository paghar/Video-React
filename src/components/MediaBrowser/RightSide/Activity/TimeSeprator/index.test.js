//@flow
import React from 'react';
import TimeSeprator from '.';
import renderer from 'react-test-renderer';

describe('<TimeSeprator />', () => {
  it('checks the UI', () => {
    const wrapper = renderer
      .create(<TimeSeprator time={1570857690000} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
