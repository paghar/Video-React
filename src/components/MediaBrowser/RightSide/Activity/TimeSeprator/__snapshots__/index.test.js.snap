// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeSeprator /> checks the UI 1`] = `
.c0 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  background: none;
  outline: none;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

<span
  className="c0 c1"
>
  Sat.oct 12
</span>
`;
